### PR TITLE
Fix tray icon

### PR DESCRIPTION
**Pull request by dusansimic**
Tray icons was not turning into `unread` state when new messages are
received. There was a bug for group chat photo because it uses
background image on div instead img tag. Also `updateBadge()` method was
inside if statement that is true only on darwin platform so tray icon
wasn't changed on linux/win32.

Closes: https://github.com/sindresorhus/caprine/issues/578
Closes: https://github.com/sindresorhus/caprine/issues/582
Closes: #594

### CODE REVIEW

**Comment by piomar123** [comment]
I tested it under Windows 7 and the icon still doesn't change on new messages.

**Comment by remmyle1102** [comment]
when will this bug be fixed? :(

**Comment by yermulnik** [comment]
When this is going to be released?

**Comment by dusansimic** [comment]
cc @sindresorhus

File: browser.js [hunk 1]
```diff
@@ -333,8 +333,9 @@ async function sendConversationList() {
 				selected: el.classList.contains('_1ht2'),
 				unread: el.classList.contains('_1ht3'),
 				icon: await getDataUrlFromImg(
-					el.querySelector('._55lt img'),
-					el.classList.contains('_1ht3')
+					el.querySelector('._55lt img') ? el.querySelector('._55lt img') : el.querySelector('._4ld- div'),
```
Comment (sindresorhus) [inline browser.js]: `el.querySelector('._55lt img')` should be a variable with a descriptive name as it's used multiple times.

Comment (sindresorhus) [reply browser.js]: `el.querySelector('._4ld- div')` should also be a descriptive variable as it's not clear what it is when just reading the code.

File: browser.js [hunk 2]
```diff
@@ -389,6 +390,9 @@ function getDataUrlFromImg(img, unread) {
 			return resolve(img.dataUnreadUrl);
 		}
 
+		if (!isImg) {
+			img.src = img.style.backgroundImage.substring(5, img.style.backgroundImage.length - 2);
+		}
```
Comment (sindresorhus) [inline browser.js]: This logic should not be in this function. This function is purely to make a data URl from an image. Move it to somewhere around line 327.

File: browser.js [hunk 3]
```diff
@@ -389,6 +390,9 @@ function getDataUrlFromImg(img, unread) {
 			return resolve(img.dataUnreadUrl);
 		}
 
+		if (!isImg) {
+			img.src = img.style.backgroundImage.substring(5, img.style.backgroundImage.length - 2);
```
Comment (sindresorhus) [inline browser.js]: Use `slice` instead of `substring` and add a short comment on what this does.

File: browser.js [hunk 4]
```diff
@@ -389,6 +390,9 @@ function getDataUrlFromImg(img, unread) {
 			return resolve(img.dataUnreadUrl);
 		}
 
+		if (!isImg) {
```
Comment (sindresorhus) [inline browser.js]: `isImg` => `isImage`

**Comment by sindresorhus** [comment]
Thanks for fixing, @dusansimic :)

**Pull request merged**